ipv6 causes crash on ubuntu

If you compile and run the latest source on ubuntu xenial it crashes on ipv6 sockets:
```
0x00000000004014f6 in protocol_to_string (proto=0) at sockstat.c:256
256             return type[proto - 1];
(gdb) bt
#0  0x00000000004014f6 in protocol_to_string (proto=0) at sockstat.c:256
#1  0x00000000004019b8 in display_record (record=0x7ffff7fa2468, pid=1908, pname=0x604240 <ret> "loopgame") at sockstat.c:342
#2  0x0000000000402db6 in main (argc=1, argv=0x7fffffffe4f8) at sockstat.c:664
```
Compilation warns about the related argument sizes.
```
make clean all
rm -f *.o sockstat
cc    -c -o sockstat.o sockstat.c
sockstat.c: In function ‘read_proc_net’:
sockstat.c:383:20: warning: format ‘%hx’ expects argument of type ‘short unsigned int *’, but argument 7 has type ‘unsigned char *’ [-Wformat=]
    if (sscanf(buf, "%*d: %lX:%x %lX:%x %hx %*X:%*X %*X:%*X %*x %u %*u %u",
                    ^
sockstat.c:383:20: warning: format ‘%u’ expects argument of type ‘unsigned int *’, but argument 9 has type ‘ino_t * {aka long unsigned int *}’ [-Wformat=]
sockstat.c:397:21: warning: format ‘%hx’ expects argument of type ‘short unsigned int *’, but argument 37 has type ‘unsigned char *’ [-Wformat=]
     if (sscanf(buf, "%*d: %2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx:%x %2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2
                     ^
sockstat.c:397:21: warning: format ‘%u’ expects argument of type ‘unsigned int *’, but argument 39 has type ‘ino_t * {aka long unsigned int *}’ [-Wformat=]
sockstat.c:442:23: warning: format ‘%u’ expects argument of type ‘unsigned int *’, but argument 7 has type ‘unsigned char *’ [-Wformat=]
     count=sscanf(buf, "%*x: %u %u %u %u %u %u %s",
                       ^
sockstat.c:442:23: warning: format ‘%u’ expects argument of type ‘unsigned int *’, but argument 8 has type ‘ino_t * {aka long unsigned int *}’ [-Wformat=]
sockstat.c:442:23: warning: format ‘%s’ expects argument of type ‘char *’, but argument 9 has type ‘char * (*)[1024]’ [-Wformat=]
sockstat.c:451:35: warning: format ‘%s’ expects argument of type ‘char *’, but argument 4 has type ‘char **’ [-Wformat=]
     snprintf(netdata[i].path,1024,"%s",path);
                                   ^
sockstat.c:451:35: warning: format ‘%s’ expects argument of type ‘char *’, but argument 4 has type ‘char **’ [-Wformat=]
cc  -o sockstat  sockstat.o
```

--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@ SRCS = sockstat.c
 OBJS = ${SRCS:.c=.o}
 
 sockstat: ${OBJS}
-	$(CC) -o $@ $(LDFLAGS) ${OBJS}
+	$(CC) $(CFLAGS) -o $@ $(LDFLAGS) ${OBJS}
 
 install:
 	install -m 755 -c sockstat ${DESTDIR}/usr/bin

--- a/sockstat.c
+++ b/sockstat.c
@@ -356,14 +356,13 @@ unsigned int read_proc_net(char initialFileToRead)
 	char protocol;
 
 	netdata = xcalloc(sizeof(procnet_entry_t), size);
-
 	while ((protocol = read_tcp_udp_raw(buf, sizeof(buf), initialFileToRead)) != 0)
 	{
 		char *q, *x, *y;
 		int lport, rport;
 		uid_t uid;
 		ino_t inode;
-		unsigned char status;
+		unsigned short status;
 		/* for unix records */
 		int refcount;
 		int proto;
@@ -377,10 +376,9 @@ unsigned int read_proc_net(char initialFileToRead)
 			size *= 2;
 			netdata = xrealloc(netdata, (sizeof(procnet_entry_t) * size));
 		}
-
 		if (protocol <= PROTOCOL_MAX_V4) {
 			/* we have an V4 entry */
-			if (sscanf(buf, "%*d: %lX:%x %lX:%x %hx %*X:%*X %*X:%*X %*x %u %*u %u",
+			if (sscanf(buf, "%*d: %lX:%x %lX:%x %hx %*X:%*X %*X:%*X %*x %u %*u %lu",
 				   (u_long *) &netdata[i].local_addr, &lport,
 				   (u_long *) &netdata[i].remote_addr, &rport, &status, 
 				   &uid, &inode) != 7)
@@ -394,7 +392,7 @@ unsigned int read_proc_net(char initialFileToRead)
 		} else {
 			if ((protocol > PROTOCOL_MAX_V4) && (protocol <= PROTOCOL_MAX_V6)) {
 				/* we have an V6 entry */
-				if (sscanf(buf, "%*d: %2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx:%x %2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx:%x %hx %*X:%*X %*X:%*X %*x %u %*u %u",
+				if (sscanf(buf, "%*d: %2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx:%x %2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx:%x %hx %*X:%*X %*X:%*X %*x %u %*u %lu",
 				   &netdata[i].local6_addr.s6_addr[ 3],
 				   &netdata[i].local6_addr.s6_addr[ 2],
 				   &netdata[i].local6_addr.s6_addr[ 1],
@@ -439,16 +437,16 @@ unsigned int read_proc_net(char initialFileToRead)
 				netdata[i++].protocol = protocol;
 			} else {
 				/* we have an unix entry */
-				count=sscanf(buf, "%*x: %u %u %u %u %u %u %s",
+				count=sscanf(buf, "%*x: %u %u %u %u %hx %lu %s",
 					&refcount,
 					&proto,
 					&flags,
 					&type,
 					&status,
 					&inode,
-					&path);
+					(char*)&path);
 				if ((count!=6) && (count!=7)) continue;
-				snprintf(netdata[i].path,1024,"%s",path);
+				snprintf(netdata[i].path,1024,"%s",(char*)path);
 				netdata[i].refcount=refcount;
 				netdata[i].proto=proto;
 				netdata[i].flags=flags;

